src/Common/ThreadPool.cpp

```diff
@@ -819,6 +826,9 @@ void ThreadPoolImpl<Thread>::ThreadFromThreadPool::worker()
             job_data.reset();
         }
 
+        DB::Exception::parent_thread_id.store(0, std::memory_order_relaxed);
```

Comment:
Related to comment above, how about just use `memory_order_release` without `atomic_signal_fence`?
```suggestion
        DB::Exception::parent_thread_id.store(0, std::memory_order_release);
```